soh/soh/Enhancements/randomizer/location_access.cpp

```diff
@@ -210,12 +210,199 @@ bool Here(const RandomizerRegion region, ConditionFn condition) {
+    {RR_SPIRIT_TEMPLE_MQ_SUN_BLOCK_ROOM,       SpiritLogicData(7, 7, 0, []{return logic->CanHitSwitch() && logic->MQSpiritStatueToSunBlock()/* && logic->CanClimbHigh()*/;}, []{return logic->MQSpiritStatueToSunBlock()/* && logic->Climb*/;},                                             []{return logic->MQSpiritStatueToSunBlock()/* && (logic->CanClimb() || logic->CanUse(RG_HOVER_BOOTS))*/;})},
+    {RR_SPIRIT_TEMPLE_MQ_BIG_BLOCK_ROOM_NORTH, SpiritLogicData(7, 7, 0, []{return logic->CanHitSwitch() && 
+                                                                        areaTable[RR_SPIRIT_TEMPLE_MQ_BIG_BLOCK_ROOM_NORTH].Here([]{return logic->MQSpiritStatueSouthDoor();})
+                                                                        /* && logic->CanClimbHigh()*/;},                                                                     []{return true;},                                                                                              []{return areaTable[RR_SPIRIT_TEMPLE_MQ_BIG_BLOCK_ROOM_NORTH].Here([]{return logic->MQSpiritStatueSouthDoor();});})},
+};
+// clang-format on
+
+bool SpiritCertainAccess(RandomizerRegion region) {
+    SpiritLogicData& curRegionData = Region::spiritLogicData[region];
+    uint8_t keys = curRegionData.adultKeys;
+    bool reverseAccess = logic->ReverseSpiritAdult;
+    if (logic->IsChild) {
+        // If child enters in reverse, then they have access to Certain Access to Broken Wall room in 6 keys,
+        // the ability to hit switches and the ability to climb because only child can reach the initial child lock
+        // without opening the Statue room to Broken Wall Room lock first
+        keys = (logic->ReverseSpiritChild && logic->CanHitSwitch() /* && CanClimbHigh()*/)
+                   ? curRegionData.childReverseKeys
+                   : curRegionData.childKeys;
+        reverseAccess = logic->ReverseSpiritChild;
+    }
+    // If we have enough keys that an age cannot be kept out, we have Certain Access
+    // otherwise if we have entered in reverse and can reach from the face, we have Certain Access
+    return logic->SmallKeys(RR_SPIRIT_TEMPLE, keys) || (reverseAccess && curRegionData.reverseAccess());
 }
 
-bool MQSpiritSharedBrokenWallRoom(const RandomizerRegion region, ConditionFn condition, bool anyAge) {
-    return areaTable[region].MQSpiritShared(condition, true, anyAge);
+/*
+    * Spirit Shared can take up to 3 regions, this is because checks can exist in many regions at the same time
+    and the logic needs to be able to check the access logic from those regions to check the other universes properly.
+
+    *anyAge is equivalent to a self referencing Here, used for events and any check where that is relevent.
+*/
+
+bool SpiritShared(RandomizerRegion region, ConditionFn condition, bool anyAge, RandomizerRegion otherRegion,
+                  ConditionFn otherCondition, RandomizerRegion thirdRegion, ConditionFn thirdCondition) {
```

Comment:
seems like this wants to be `(bool anyAge, initializer-list-of-pairs)` but this works for 3 for now